### DISCUSSION TITLE
tägliches Programmieren in Python

### DISCUSSION THREAD

**Discussion by DGrillCoder** [grilldb97/legendary-space-zebra#3] [Announcements]
Start:
05.12.2023: gestartet mit der Python-Programmierung in GitHub
06.12.2023: Variablen
07.12.2023: Input
08.12.2023: Datentypen geprüft
09.12.2023: Rechnen
10.12.2023: Vergleichsoperatoren
11.12.2023: Schleifen

kleine Programme:
12.12.2023: Ein Programm zur Wahl von Personen und dessen Wahlergebnis
13.12.2023: Programm zur Berechnung des Benzinverbrauchs
14.12.2023: objektorientiert - HelloWorld
15.12.2023: Würfelspiel - mit random und time
16.12.2023: meinen Start(s.o.) in Python ergänzt
17.12.2023: Uhrzeit - Grüße: Anhand der Eingabe der Uhrzeit wird gegrüßt

Projekt: Wecker
18.12.2023: Wecker: Anfang eines Weckers - Ausgabe der Uhrzeit
19.12.2023: Wecker: erweitert mit Eingabe der Weckzeit und Ausgabe eines Beep
20.12.2023 -Wecker: umstrukturiert mit Funktionen; erweitert mit grafischer Oberfläche zur Eingabe der Weckzeit, farbiger Anzeige der Uhrzeit und Ausgabe von Ton
21.12.2023: Wecker: Uhrzeit in GUI integriert. Eingabefeld von Stunden:Minuten in Spinboxen gesetzt
22.12.2023: Wecker: mehrere Weckzeiten in den Wecker implementiert; Buttons: Snooze/Stop; Wecker löschen implementiert; aktuell noch das Problem, dass der Alarm nicht funktioniert
23.12.2023: Wecker: Buttons: Snooze/Stop getrennt; implementiert, dass man zw. Alarm und Musik wechseln kann; aktuell noch das Problem, dass der Alarm nicht funktioniert und die Tabs nicht blinken

24.12.2023: Christmas Tree (https://www.codetoday.co.uk/post/how-to-draw-a-christmas-tree-using-turtle-in-python)
25.12.2023: Christmas Tree erweitert mit dem Nikolaushaus
26.12.2023: Christmas Tree: der Baum wird geschmückt und es hat angefangen zu schneien. Es bleibt sogar liegen. Frohe Weihnachten :)

27.12.2023: Wecker: Funktionen in Klassen gesetzt: Alarm, Wecker, Window. Das Hauptprogramm ruft lediglich nur noch das Fenster' WECKER' auf => weitere Analyse folgt
28.12.2023: Wecker umstrukturiert - nicht funktionsfähig
29.12.2023: Wecker weiter funktionsfähig zu machen
30.12.2023: Wecker nochmals neu aufgebaut. Schritt für Schritt die GUI definiert und anschließend die Funktionen definiert
31.12.2023: Wecker: probiert weitere Funktionen zu implementieren. Aktuell hänge ich an wecker_stellen

01.01.2024: Wecker: Alarm nun funktionsfähig
02.01.2024: Wecker: Index auf wecker_index gesetzt und im kompletten Code angeglichen. Aktuell ein Problem mit den Spinboxes. Vmtl. da diese static ist. Weitere Analyse notwendig
03.01.2024: Wecker: Wecker läuft wieder. static in create_spinbox entfernt. Der wecker_index beim wecker_stellen war falsch. Code so angepasst, dass der button_wecker_stellen nun selbst prüft, in welchem Tab er ist und dies als index weiter gibt. Aktuell wird kein Alarm ausgegeben. Weiter analysieren
04.01.2024: Wecker: Weiter analysieren
05.01.2024: Wecker: alarm_queues umstrukturiert; Threads verschoben
06.01.2024: Wecker: weiter analysiert. Alarm funktioniert aktuell, nur in Wecker1. 

07.01.2024: Galgenmännchen: mit Verwendung einer Liste, random, while, for, if/else habe ich heute zur Abwechslung ein Galgenmännchen-Spiel gemacht

08.01.2023: Datenbank: eine Datenbank mit Modul sql3lite angelegt
09.01.2024: Datenbank: Programm erweitert mit Benutzereingabe, Schreiben in die Datenbank und Datenbankinhalt auslesen
10.01.2024: Datenbank: erweitert mit einer Suche für die Datenbank
11.01.2024: Datenbank: Suche in einer Funktion ausgelagert; erweitert mit der Aktualisierung von Datensätzen; 
12.01.2024: Datenbank: Bearbeiten eines Datensatzes überarbeitet; Programm erweitert mit dem Löschen eines Datensatzes
13.01.2024: Android App: Start mit einer Android App mit Kivy
14.01.2024: Android App: datenbank1 angefangen in einer App darzustellen
15.01.2024: Android App: Abfrage beim Start der App, ob jemand Geburtstag hat
16.01.2024: Android App: Suche implementiert

17.01.2024: Triangles: Dreiecke werden nach Eingabe der Zeilenanzahl ausgegeben
18.01.2024: Pyramide: wird nach Eingabe einer Zahl ausgegeben
19.01.2024: rock, paper, scissor: Schere, Stein, Papier erstellt
20.01.2024: turtle_labyrinth: einfaches quadratisches Labyrinth
21.01.2024: list: mit Listen in Python beschäftigt
22.01.2024: get_grade: if, elif, else

23.01.2024: kivy: BoxLayout
24.01.2024: kivy: AnchorLayout
25.01.2024: kivy: GridLayout
26.01.2024: kivy: StackLayout
27.01.2024: kivy: ScrollView mit StackLayout
28.01.2024: kivy: PageLayout - mit dem Finger nach links wischen und dann kommen die einzelnen Seiten

29.01.2024: kivy: ButtonClick
30.01.2024: kivy: ClickCounter
31.01.2024: kivy: ToggleButton - Umschalttaste
01.02.2024: kivy: Toggle the ClickCounter (on/off)
02.02.2024: kivy: Switch-Button

03.02.2024: kivy: Slider
04.02.2024: kivy: SwitchSlider
05.02.2024: kivy: ProgressBar

06.02.2024: kivy: TextInput
07.02.2024: kivy: TextInput bei Enter + Label

08.02.2024: kivy: Images

09.02.2024: kivy: Canvas - Rectangle + Ellipse
10.02.2024: kivy: Canvas - Line
11.02.2024: kivy: Canvas - Line: circle - ellipse - rectangle
12.02.2024: kivy: Canvas - Move the rectangle
13.02.2024: kivy: Canvas - Ball Animation
14.02.2024: kivy: Canvas - RelativeLayout

15.02.2024: kivy: Galaxy Game App - vertical lines
16.02.2024: kivy: Galaxy Game App - Perspective transformation
17.02.2024: kivy: Galaxy Game App - Horizontal lines + Horitonzal lines perspective
18.02.2024: kivy: Galaxy Game App - Forward movement
19.02.2024: kivy: Galaxy Game App - Delta time (time_factor in Methode update
20.02.2024: kivy: Galaxy Game App - Lateral movement; Key press
21.02.2024: kivy: Galaxy Game App - Keyboard

22.02.2024: kivy: Galaxy Game App (V2) - Structure our Code, Tiles and lines coordinates, Line Y from index
23.02.2024: kivy: Galaxy Game App (V2) - Display a tile; move the tile; Display several tiles
24.02.2024: kivy: Galaxy Game App (V2) - Tiles: infinite generation; Random land generation; 
                                                                   Do not exit the land; start with a straight line; 
                                                                   Display the ship; Fix the game speed; Collisions

25.02.2024: kivy: Galaxy Game App (V3) - "Game over" state; Create the menu; 
                                                                    Start of the game; Restart the game; 
                                                                    Finalise the menu; add the background image; 
                                                                    Score; add the sound;

**Comment by SirPatschiii** [comment]
Schöne Schritt für Schritt Aufarbeitung von den verschiedenen Themen.
Kleiner Tipp zum 12.12.2023: mit der round() Methode macht man sich beim Output das Leben etwas einfacher.
Hierfür einfach die Variable übergeben und die Anzahl der Nachkommastellen.
Beispiel: round(4.5698, 2) ergibt dann 4.57

**Comment by JonNoe** [comment]
Finde die tägliche Arbeit von dir super 👍 Mach weiter so
